File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py

def solution(arr):
    answer = []
    prev = -1
    for i in range(len(arr)):
        if prev == -1:
            answer.append(arr[i])
            prev = arr[i]
        else:
            if prev == arr[i]:
                continue
            else:
                answer.append(arr[i])
                prev = arr[i]
    return answer

File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        int prev = -1;
        for(int i = 0; i < arr.length; ++i){
            if(prev == -1){
                prev = arr[i];
                answer.add(arr[i]);
            }else{
                if(prev == arr[i])
                    continue;
                else{
                    prev = arr[i];
                    answer.add(arr[i]);
                }
            }
        }


        return answer.stream()
	.mapToInt(Integer::intValue)
    	.toArray();
    }
}